src/lsptoolshost/buildDiagnosticsService.ts

```diff
@@ -0,0 +1,149 @@
+        const buildDiagnosticsToDisplay: vscode.Diagnostic[] = [];
+
+        // If it is a project system diagnostic (e.g. "Target framework out of support")
+        // then always show it. It cannot be reported by live.
+        const projectSystemDiagnostics = diagnosticList.filter((d) =>
+            BuildDiagnosticsService.isProjectSystemDiagnostic(d)
+        );
+        buildDiagnosticsToDisplay.push(...projectSystemDiagnostics);
+
+        // If it is a "build-only"diagnostics (i.e. it can only be found by building)
+        // then always show it. It cannot be reported by live.
+        const buildOnlyDiagnostics = diagnosticList.filter((d) =>
+            BuildDiagnosticsService.isBuildOnlyDiagnostic(buildOnlyIds, d)
+        );
+        buildDiagnosticsToDisplay.push(...buildOnlyDiagnostics);
+
+        // Check the analyzer diagnostic setting. If the setting is "none" or if the file is closed,
+        // then no live analyzers are being shown and bulid analyzers should be added.
+        // If FSA is on, then this is a no-op as FSA will report all analyzer diagnostics
+        if (
+            analyzerDiagnosticScope === AnalysisSetting.None ||
+            (analyzerDiagnosticScope === AnalysisSetting.OpenFiles && !isDocumentOpen)
+        ) {
+            const analyzerDiagnostics = diagnosticList.filter(
+                // Needs to be analyzer diagnostics and not already reported as "build only"
+                (d) => BuildDiagnosticsService.isAnalyzerDiagnostic(d) && !this.isBuildOnlyDiagnostic(buildOnlyIds, d)
+            );
+            buildDiagnosticsToDisplay.push(...analyzerDiagnostics);
+        }
+
+        // Check the compiler diagnostic setting. If the setting is "none" or if the file is closed,
+        // then no live compiler diagnostics are being shown and bulid compiler diagnostics should be added.
+        // If FSA is on, then this is a no-op as FSA will report all compiler diagnostics
+        if (
+            compilerDiagnosticScope === AnalysisSetting.None ||
+            (compilerDiagnosticScope === AnalysisSetting.OpenFiles && !isDocumentOpen)
+        ) {
+            const compilerDiagnostics = diagnosticList.filter(
+                // Needs to be analyzer diagnostics and not already reported as "build only"
+                (d) => BuildDiagnosticsService.isCompilerDiagnostic(d) && !this.isBuildOnlyDiagnostic(buildOnlyIds, d)
+            );
+            buildDiagnosticsToDisplay.push(...compilerDiagnostics);
+        }
+
+        return buildDiagnosticsToDisplay;
+    }
+
+    private static isBuildOnlyDiagnostic(buildOnlyIds: string[], d: vscode.Diagnostic): boolean {
+        return buildOnlyIds.find((b_id) => b_id === d.code) !== undefined;
+    }
+
+    private static isCompilerDiagnostic(d: vscode.Diagnostic): boolean {
+        // eslint-disable-next-line prettier/prettier
+        const regex = "[cC][sS][0-9]{4}";
```

Comment:
what is the prettier warning here?

Reply:
Good catch, it just had a problem with the double quote "